src/ProjectService.Validation/EditProjectValidator.cs

```diff
@@ -11,67 +11,93 @@
-                        RuleFor(x => x.Patch.Operations)
-                        .UniqueOperationWithAllowedOp(NamePath, "add", "replace");
+                        RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(Description, "replace");
+                    });
 
-                        RuleFor(x => (string)GetOperationByPath(x.Patch, NamePath).value)
-                        .NotEmpty()
-                        .WithMessage($"{NamePath} must not be empty.")
-                        .MaximumLength(80)
-                        .WithMessage($"{NamePath} is too long.");
+                    When(x => GetOperationByPath(x, ShortDescription) != null, () =>
+                    {
+                        RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(ShortDescription, "replace");
+
+                        RuleFor(x => (string)GetOperationByPath(x, ShortDescription).value)
+                            .MaximumLength(300).WithMessage("Short description is to long");
                     });
 
-                    When(x => GetOperationByPath(x.Patch, ShortNamePath) != null, () =>
+
+                    When(x => GetOperationByPath(x, Status) != null, () =>
                     {
-                        RuleFor(x => x.Patch.Operations)
-                        .UniqueOperationWithAllowedOp(ShortNamePath, "add", "replace", "remove");
+                        RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(Status, "replace");
 
-                        When(x => GetOperationByPath(x.Patch, ShortNamePath).op != "remove", () =>
-                        {
-                            RuleFor(x => (string)GetOperationByPath(x.Patch, ShortNamePath).value)
-                            .MaximumLength(32)
-                            .WithMessage($"{ShortNamePath} is too long.");
-                        });
+                        RuleFor(x => (ProjectStatusType)GetOperationByPath(x, Status).value)
+                            .IsInEnum().WithMessage("Wrong status value.");
                     });
 
-                    When(x => GetOperationByPath(x.Patch, DescriptionPath) != null, () =>
+                    When(x => GetOperationByPath(x, DepartmentId) != null, () =>
                     {
-                        RuleFor(x => x.Patch.Operations)
-                        .UniqueOperationWithAllowedOp(DescriptionPath, "add", "replace", "remove");
+                        RuleFor(x => x.Operations).UniqueOperationWithAllowedOp(DepartmentId, "replace");
 
-                        When(x => GetOperationByPath(x.Patch, DescriptionPath).op != "remove", () =>
-                        {
-                            RuleFor(x => (string)GetOperationByPath(x.Patch, DescriptionPath).value)
-                            .MaximumLength(500)
-                            .WithMessage($"{DescriptionPath} is too long.");
-                        });
+                        RuleFor(x => GetOperationByPath(x, DepartmentId).value)
```

Comment:
Where is check for existing in DB?

Reply:
In EditProjectCommand i send a request throw broker to be shore what the department Id is exist

Reply:
I think it would be better move to validator, but ok

Reply:
I use it in command because in the command i have the same request to CompanyService to get DepartmentDirector, in this way i can reuse GetDepartment method